### DISCUSSION TITLE
ROS2 connection drops after inactivity

### DISCUSSION THREAD

**Discussion by vivianweli** [iRobotEducation/create3_docs#337] [Questions]
I'm able to connect to the robot, display items from ros2 topic list etc and run packages. After a while of inactivity (e.g. 1 hour), I lose connectivity with the robot. ros2 topic list no longer returns only /ros_out and /parameter_events. What am I missing here? Or is there a command I could do to re-establish connection? When I reboot the robot, then connection is re-established, but doing this every time is impractical.
My machine is running Ubuntu 22.04 on Virtualbox (bridged connection). I'm using humble and the robot's firmware is H.0.0.

**Comment by shamlian** [comment]
A few thoughts without solid answers:
1) I'd recommend upgrading to H.1 from H.0. Download the firmware from the web and upload it to the robot. 
2) If this happens after upgrading, please download the raw logs from the robot web server and post them. It would also be helpful to know more about your network, middleware choice, etc.

**Reply by richardw347** [reply]
I've also been having this issue. I've tried both H.1.0 and G.4.3. I've attached a log file from the most recent time it's happened (my robot is running G.4.3 and I'm using cyclone-dds. I'm getting a lot of the same messages when I can't see the topics: 

```
Mar  2 14:19:03 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766743.913876 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:11 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766751.913981 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:13 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 daemon.notice network[1464]: [INFO] (src/gateway_pinger.c:113) wifi is disconnected
Mar  2 14:19:19 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766759.914087 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:27 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766767.914639 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:33 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 daemon.notice network[1464]: [INFO] (src/gateway_pinger.c:113) wifi is disconnected
Mar  2 14:19:35 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766775.914737 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:43 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766783.914867 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:51 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 user.notice create-platform: 1677766791.914994 [0]        tev: ddsi_udp_conn_write to udp/239.255.0.1:7400 failed with retcode -1
Mar  2 14:19:53 iRobot-A01664DEBB0F46C0BAE86BF0D937FEF8 daemon.notice network[1464]: [INFO] (src/gateway_pinger.c:113) wifi is disconnected
```

I've attached the full log. Note after ~15:30 in the logs is where I restart the robot application via the Web UI. 

[messages.txt](https://github.com/iRobotEducation/create3_docs/files/10873138/messages.txt)

**Reply by shamlian** [reply]
Thanks for this @richardw347 . Can you tell me a little bit about how your robot is connected to networks? Just Wi-Fi, just usb0, or both? Looks like just usb0, but I want to make sure that I fully understand the intent for the network environment. I have a suspicion about why this might be happening, so thank you again for posting this log.

**Reply by richardw347** [reply]
Hi @shamlian I'm just using usb0. Let me know if any further logs/test would be helpful.

**Reply by shamlian** [reply]
Off the top of my head, I'm curious to know if this issue is new in G.4 vs G.3; if you are willing and able to see if you can reproduce in G.3 that could be helpful. I am working to reproduce your issue locally, as well, so if you are not able, hopefully I can. In the short term while we figure this out, if you do stay on G.4/H.1, you can `curl -X POST http://192.168.186.2/api/reboot` to quickly bonk the robot without having to muck with the dock and power button.

**Reply by richardw347** [reply]
I'll switch over to G.3 and try it this weekend. Thanks for the tips about the reboot I might add a quick node to check topics reboot!
